src-worker/services/configurationService.ts

```diff
@@ -0,0 +1,67 @@
+import { GetSitesAndProxiesResponse, ProxyData, SiteData } from '../types'
+import { getProxies, getSites } from '../requests'
+
+export class ConfigurationService {
+  public static readonly CONFIGURATION_INVALIDATION_TIME = 300000 // 5 min
+  private proxies: ProxyData[] | undefined = undefined
+  private sites: SiteData[] | undefined = undefined
+  private expiredAt: Date | undefined = undefined
+
+  async pullConfiguration (): Promise<GetSitesAndProxiesResponse> {
+    if ((this.proxies && this.sites) || this.expired()) {
+      console.log('Reset configs')
+      this.proxies = undefined
+      this.sites = undefined
+      this.expiredAt = undefined
+    }
+
+    let valid = true
+    if (!this.sites) {
+      try {
+        const sites = await getSites()
+        if (sites.status === 200) {
+          this.sites = sites.data
+          this.updateExpirationDate()
+        } else {
+          valid = false
+        }
+      } catch (e) {
+        valid = false
+      }
+    }
+    if (!this.proxies) {
+      try {
+        const proxies = await getProxies()
+        if (proxies.status === 200) {
+          this.proxies = proxies.data
+          this.updateExpirationDate()
+        } else {
+          valid = false
+        }
+      } catch (e) {
+        valid = false
+      }
+    }
+
+    if (!valid) {
+      throw Error('Can not pull configuration')
+    }
+
+    return {
+      sites: (this.sites as SiteData[]),
+      proxies: (this.proxies as ProxyData[])
```

Comment:
avoid type casting here, you can declare props as follows

`private sites!: SiteData[]`

Here returning the state variables would most likely cause unexpected side effects, try to return what you got inside the method, 
`const sites = await getSites()`
`return { sites: sites.data }` -> no need to type cast

if you want to share this properties externally make them `public` or just remove `private` keyword for simplicity.

Just an idea  one thing you could do is.

```js
getConfig() { 
  return { sites: this.sites, proxies: this.proxies }
}
```
outside check:
```js
const { sites, proxies } = config.getConfig()
config.pullConfiguration (!!sites, !!proxies) // You cast whatever the result is (undefined | object) to boolean.
// this way you know exactly what to pull
```